### PR TITLE
Divyanshu | Test | Version 3 - Get all voucher page

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cupmt5p

### CODE REVIEW

File: apps/web-giddh/src/app/services/http-wrapper.service.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export class HttpWrapperService {
         options.headers["Session-Id"] = this.generalService.sessionId;
         options.headers["Content-Type"] = "application/json";
         options.headers["Accept"] = "application/json";
-        options.headers["X-Tenant"] = this.generalService.getUtmParameter("X-Tenant");
+        // options.headers["X-Tenant"] = this.generalService.getUtmParameter("X-Tenant");
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/http-wrapper.service.ts]: Confirm with @dilpreetsinghdang  sir

File: apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.html [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
+                    formControlName="purchaseNumber"
+                >
+                </input-field>
+                <span class="ml-1">{{ form?.get("purchaseOrders")?.value[i]?.orderNumber }}</span>
+            </div>
+        </ng-container>
+    </mat-dialog-content>
+    <mat-dialog-actions>
+        <div class="d-flex column-gap1 justify-content-end mt-4">
+            <button mat-stroked-button color="primary" type="submit">Save</button>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.html]: translation

File: apps/web-giddh/src/app/vouchers/create/create.component.ts [hunk 3]
```diff
@@ -3731,6 +3731,7 @@ export class VoucherCreateComponent implements OnInit, OnDestroy, AfterViewInit
                     }
                 });
             } else {
+                console.log(invoiceForm);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/create/create.component.ts]: remove

File: apps/web-giddh/src/app/vouchers/create/create.component.ts [hunk 4]
```diff
@@ -3879,6 +3880,8 @@ export class VoucherCreateComponent implements OnInit, OnDestroy, AfterViewInit
...
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/create/create.component.ts]: remove

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 5]
```diff
@@ -1,25 +1,37 @@
...
             
             <!-- estimate-content-start -->
-            <mat-tab label="Estimates (Beta)">
+            <mat-tab label="Estimates (Beta)" *ngIf="activeTabGroup === 0">
                 <div class="vouchers-wrapper pt-3 px-4">
                     <div class="invoice-header d-flex align-items-center justify-content-between mr-b15">
                         <div class="d-flex align-items-center column-gap1">
-                            <button mat-stroked-button color="primary">
+                            <button mat-stroked-button color="primary" [routerLink]="['/pages/vouchers/estimates/create']">
                                 <span>+ New Estimate</span>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: translation

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 6]
```diff
@@ -32,201 +44,228 @@
...
+                                <td mat-cell *matCellDef="let element"> 
+                                    {{ element.status }}
                                 </td>
                             </ng-container>
                             <ng-container matColumnDef="action">
                                 <th mat-header-cell *matHeaderCellDef>
                                     Action
                                 </th>
                                 <td mat-cell *matCellDef="let element"> 
+                                    <!-- *ngIf="element.action !== 'invoiced'" ========== NEED TO ASK WHY THIS -->
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: ????

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 7]
```diff
@@ -32,201 +44,228 @@
...
+                                        name="selectedDateRange"
+                                        [value]="selectedDateRangeUi ? selectedDateRangeUi : ''"
+                                        class="giddh-datepicker date-range-picker border-radius-2"
+                                    />
                                 </div>
                             </div>
                         </div>
                         <div class="d-flex align-items-center column-gap1">
-                            <button mat-stroked-button color="primary" [matMenuTriggerFor]="moreProforma">
+                            <!-- ======================= Ask what is use of this ======================= -->
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: ????

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 8]
```diff
@@ -734,58 +829,86 @@ <h3>00000000</h3>
...
                                     <div class="d-flex justify-content-between align-items-center column-gap5">
-                                        <p>Cancel</p>
-                                        <span [matMenuTriggerFor]="menu" class="cp">
+                                        <p>{{ element?.balanceStatus | titlecase }}</p>
+                                        <span [matMenuTriggerFor]="menu" class="cursor-pointer">
                                             <i class="icon-dots-three-vertical"></i>
                                         </span>
                                         <mat-menu #menu="matMenu">
-                                            <button mat-menu-item>Cancel</button>
+                                            <button mat-menu-item (click)="showPaymentDialog(element)">Paid</button>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: translation

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 9]
```diff
@@ -1,15 +1,214 @@
...
+        baseCurrency: '',
+        baseCurrencySymbol: '',
+        inputMaskFormat: '',
+        giddhBalanceDecimalPlaces: 0
+    };
+    public tags: any[] = [];
+    /** Brief accounts Observable */
+    public briefAccounts$: Observable<OptionInterface[]> = observableOf(null);
+    public paymentForm: FormGroup;
+    public amountCurrency: string = "";
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts:41]: JS Doc

File: apps/web-giddh/src/app/vouchers/po-bulk-update/po-bulk-update.component.html [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<p>
+  po-bulk-update works!
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/po-bulk-update/po-bulk-update.component.html]: Need to check

File: apps/web-giddh/src/assets/locale/vouchers/preview/en.json [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+    "status_updated": "Status updated successfully.",
+    "delayed_by_days": "Delayed by [DAYS] days",
+    "delivery_in_day": "delivery_in_day[DAYS] day",
+    "delivery_in_days": "Delivery in [DAYS] days",
+    "expected_delivery_date": "Expected Delivery Date",
+    "order_date": "Order Date",
+    "po_date_error": "Please select Purchase date",
+    "po_expirydate_error": "Please select Expected delivery date",
+    "po_warehouse_error": "Please select Warehouse",
+    "po_bulkupdate_error": "Please choose a field to update"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/assets/locale/vouchers/preview/en.json]: translate Hindi Marathi

File: apps/web-giddh/src/app/contact/contact.component.scss [hunk 12]
```diff
@@ -183,6 +183,10 @@ td.comment-field {
     white-space: normal;
 }
 
+.date-range-picker {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/contact/contact.component.scss]: check this

File: apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.ts [hunk 13]
```diff
@@ -325,7 +325,6 @@ export class ContentFilterComponent implements DoCheck, OnInit, OnChanges, OnDes
      */
     public changeInvoiceHeader(event: boolean): void {
         this.customTemplate.sections['header'].data['formNameInvoice'].display = event;
-        this.onChangeFieldVisibility(null, null, null);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.ts]: check this

File: apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/design-filters/design.filters.component.scss [hunk 14]
```diff
@@ -124,21 +124,6 @@
     padding: 4px;
 }
 
-.design-tool {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/design-filters/design.filters.component.scss]: check this

File: apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts [hunk 15]
```diff
@@ -90,7 +125,12 @@ export class VoucherComponentStore extends ComponentStore<VoucherState> {
         private ledgerService: LedgerService,
         private commonService: CommonService,
         private accountService: AccountService,
-        private searchService: SearchService
+        private searchService: SearchService,
+        private bulkUpdateInvoiceService: InvoiceBulkUpdateService,
+        private bulkVoucherExportService: BulkVoucherExportService,
+        private salesService: SalesService,
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts]: move methods to voucher service

File: apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts [hunk 16]
```diff
@@ -808,4 +1243,177 @@ export class VoucherComponentStore extends ComponentStore<VoucherState> {
...
+                            });
+                        }
+                    ),
+                    catchError((err) => EMPTY)
+                );
+            })
+        );
+    });
+
+    readonly POBulkUpdateAction = this.effect((data: Observable<{ payload: any, actionType: string }>) => {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts]: give proper naming

File: apps/web-giddh/src/assets/css/style-1.scss [hunk 17]
```diff
@@ -3195,62 +3197,7 @@ td.mat-footer-cell {
         }
     }
 }
-.select-physical-product {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/css/style-1.scss]: check removed css

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 18]
```diff
@@ -11043,37 +11038,6 @@ body.update-scroll-hidden {
     }
 }
 
-.ledger-attachments-popup {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/css/style-2.scss]: check removed css

File: apps/web-giddh/src/assets/locale/purchase-order/en.json [hunk 19]
```diff
@@ -70,7 +70,7 @@
     "custom_field2": "Custom field 2",
     "currency_conversion": "In [BASE_CURRENCY]: [AMOUNT]<br />(Conversion Rate: [CONVERSION_RATE])",
     "delayed_by_days": "Delayed by [DAYS] days",
-    "delivery_in_day": "Delivery in [DAYS] day",
+    "delivery_in_day": "delivery_in_day[DAYS] day",
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/purchase-order/en.json]: correct this

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 20]
```diff
@@ -1,15 +1,214 @@
...
+        private searchService: SearchService,
+        private formBuilder: FormBuilder
+    ) {
 
-  ngOnInit() {
-  }
+    }
 
-}
+    public ngOnInit(): void {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts:77]: add js doc in all methods

File: apps/web-giddh/src/app/vouchers/email-send-dialog/email-send-dialog.component.html [hunk 21]
```diff
@@ -18,7 +18,7 @@ <h4 mat-dialog-title class="font-16">{{ commonLocaleData?.app_send_email }}</h4>
                 <div class="w-100 option-value-field" formGroupName="email">
                     <text-field [name]="'email'" [placeholder]="localeData?.enter_email_addresses" [cssClass]="'w-100'" formControlName="to" [type]="'text'"></text-field>
                 </div>
-                <div class="my-3" *ngIf="!invoiceType?.isEstimateInvoice && !invoiceType?.isProformaInvoice && !invoiceType?.isDebitNote && !invoiceType?.isCreditNote && !invoiceType?.isReceiptInvoice && !invoiceType?.isPaymentInvoice">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/email-send-dialog/email-send-dialog.component.html:21]: don't remove this

File: apps/web-giddh/src/app/vouchers/email-send-dialog/email-send-dialog.component.html [hunk 22]
```diff
@@ -33,7 +33,7 @@ <h3 class="mr-b05">{{ localeData?.invoice_copy }}</h3>
                         {{ commonLocaleData?.app_customer }}
                     </mat-checkbox>
                 </div>
-                <div class="d-flex column-gap1" [ngClass]="{'mr-t1' : invoiceType?.isEstimateInvoice || invoiceType?.isProformaInvoice || invoiceType?.isDebitNote || invoiceType?.isCreditNote || invoiceType?.isReceiptInvoice || invoiceType?.isPaymentInvoice}">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/email-send-dialog/email-send-dialog.component.html:36]: don't remove this

File: apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.html [hunk 23]
```diff
@@ -1,17 +1,29 @@
...
-  >
-  </input-field>
-  <span class="ml-1">2116</span>
-  <div class="d-flex column-gap1 justify-content-end">
-    <button mat-stroked-button color="primary">Save</button>
-    <button mat-stroked-button>Cancel</button>
-  </div>
-</div>
\ No newline at end of file
+<form [formGroup]="form" autocomplete="off" novalidate *ngIf="showForm" (ngSubmit)="onFormSubmit()">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.html]: remove ngIf

File: apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts [hunk 24]
```diff
@@ -1,15 +1,82 @@
...
 })
 export class ConvertBillDialogComponent implements OnInit {
+  /* This will hold local JSON data */
+  @Input() public localeData: any = {};
+  /* This will hold common JSON data */
+  @Input() public commonLocaleData: any = {};
+  /** Emits convert To Bill form value */
+  @Output() public convertToBill = new EventEmitter<any>();
+  public form: FormGroup;
+  public showForm: boolean = true;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts]: remove var

File: apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts [hunk 25]
```diff
@@ -1,15 +1,82 @@
...
 
+  /**
+   * Set Purchase Order Value
+   *
+   * @private
+   * @param {*} voucher
+   * @return {*}  {FormGroup}
+   * @memberof ConvertBillDialogComponent
+   */
+  private setPurchaseOrderValue(voucher): FormGroup {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts]:
```suggestion
  private getPurchaseOrderValue(voucher?: any): FormGroup {
```

File: apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts [hunk 26]
```diff
@@ -1,15 +1,82 @@
...
+  public ngOnInit(): void {
+    if (this.vouchers) {
+      let formArray = this.form.get('purchaseOrders') as FormArray;
+      if (this.vouchers?.length) {
+        this.vouchers.forEach(voucher => {
+          formArray.push(this.setPurchaseOrderValue(voucher));
+        });
+      } else {
+        let formAarry = this.form.get('purchaseOrders') as FormArray;
+        formAarry.push(this.setPurchaseOrderValue(this.vouchers));
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts]: send only in array form

File: apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html [hunk 27]
```diff
@@ -1,148 +1,245 @@
...
+            </dropdown-field>
+            <input-field [type]="'tel'" [placeholder]="'00'" class="w-50">
+            </input-field>
+        </div>
+    </div> -->
     <!-- estimate-proforma-search-end -->
     <div class="d-flex justify-content-end column-gap1 mt-3">
-      <button mat-stroked-button color="primary">Search</button>
-      <button mat-stroked-button mat-dialog-close>Cancel</button>
+        <button mat-stroked-button color="primary" (click)="search()">Search</button>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html]: do translation

File: apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.ts [hunk 28]
```diff
@@ -1,45 +1,324 @@
...
+            this.fieldLabelValues.amountFieldSelector = amountFieldSelector[0]?.label;
+        }
+    }
+
+    public ngOnDestroy(): void {
+        this.destroyed$.next(true);
+        this.destroyed$.complete();
+    }
+
+    public invoiceTotalRangeChanged(item: IOption) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.ts]: add js doc and return type in all methods

File: apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.html [hunk 29]
```diff
@@ -1,83 +1,214 @@
...
+                *ngIf="bulkUpdateForm.get('selectedField')?.value === 'notes'"
+            >
+                <textarea
+                    matInput
+                    rows="3"
+                    [maxlength]="!showNotesAtLastPage ? 200 : null"
+                    formControlName="message2"
+                ></textarea>
+                <ng-container *ngIf="bulkUpdateForm.get('message2')?.value && !showNotesAtLastPage">
+                    <span class="max-character-label">Characters left:&nbsp;</span>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.html]: do translation

File: apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.ts [hunk 30]
```diff
@@ -1,15 +1,484 @@
...
+        }
+        const model = this.bulkUpdateForm.value;
+        const actionType = model.action;
+        delete model.action;
+
+        model.purchaseNumbers = this.inputData?.purchaseNumbers;
+        this.componentStore.POBulkUpdateAction({ payload: model, actionType: actionType})
+    }
+
+    public bulkUpdateRequest(payload: any, actionType: string): void {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.ts:388]: add js doc

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 31]
```diff
@@ -1,156 +1,177 @@
...
+                                        [currencySymbol]="element?.amount?.currencyForAccount?.symbol"
+                                        [currencyCode]="element?.amount?.currencyForAccount?.code"
+                                    >
+                                    </amount-field>
                                 </td>
                             </ng-container>
-                            <ng-container matColumnDef="expirydate">
+                            <ng-container matColumnDef="dueDate">
                                 <th mat-header-cell *matHeaderCellDef mat-sort-header>
                                     Expiry Date
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: translation

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 32]
```diff
@@ -159,166 +180,196 @@ <h3>00000000</h3>
...
+                            <ng-container matColumnDef="grandTotal">
                                 <th mat-header-cell *matHeaderCellDef mat-sort-header class="text-right">
-                                    Total Amt.
+                                    {{ localeData?.total_amt }}
                                 </th>
-                                <td mat-cell *matCellDef="let element" class="text-right" matTooltip="In INR: 1.239 (Conversion Rate: 0.005)"
+                                <td mat-cell *matCellDef="let element" class="text-right" [matTooltip]="element.grandTotalTooltipText"
                                 matTooltipPosition="above"> 
-                                    Rp233.00
+                                    {{ grandTotal }}
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: check this

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 33]
```diff
@@ -159,166 +180,196 @@ <h3>00000000</h3>
...
+                                    </amount-field>
                                 </td>
                             </ng-container>
-                            <ng-container matColumnDef="expirydate">
+                            <ng-container matColumnDef="dueDate">
                                 <th mat-header-cell *matHeaderCellDef mat-sort-header>
-                                    Expiry Date
+                                    {{ localeData?.expiry_date }}
                                 </th>
                                 <td mat-cell *matCellDef="let element" matTooltip="In INR: 1.239 (Conversion Rate: 0.005)"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: check this

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 34]
```diff
@@ -865,55 +1192,486 @@ <h3>USA debtor</h3>
...
-                                class="settings-field"
-                            ></input-field>
-                            <span class="text-success">Verified</span>
-                            <button mat-stroked-button class="mat-icon-md-btn">
-                                <i class="icon-trash"></i>
-                            </button>
-                        </div>
+            
+            <!-- purchase-order-start -->
+            <mat-tab label="Purchase Order" *ngIf="activeTabGroup === 2">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: do translation

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 35]
```diff
@@ -865,55 +1192,486 @@ <h3>USA debtor</h3>
...
+                            *ngTemplateOutlet="noDataFoundMessage;
+                            context: { $implicit: !dataSource?.length}"
+                        ></ng-container>
+                    </div>
+                </div>
+            </mat-tab>
+            <!-- purchase-order-end -->
+
+            <!-- bill-content-start -->
+            <mat-tab label="Bill" *ngIf="activeTabGroup === 2">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: translation

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 36]
```diff
@@ -865,55 +1192,486 @@ <h3>USA debtor</h3>
...
+                            *ngTemplateOutlet="noDataFoundMessage;
+                            context: { $implicit: !dataSource?.length}"
+                        ></ng-container>
+                    </div>
+                </div>
+            </mat-tab>
+            <!-- bill-content-end -->
+
+            <!-- setting-content-start -->
+            <mat-tab label="Settings" class="setting-svg">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: hide settings tab for now

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/services/apiurls/inventory.api.ts [hunk 37]
```diff
@@ -105,16 +105,16 @@ export const INVENTORY_API = {
     UPDATE_DISCOUNT: 'company/:companyUniqueName/customer-vendor-discount/stock/:stockUniqueName/variant/:variantUniqueName/update-discount', // Patch Call
     GET_STOCK_DETAILS: 'v2/company/:companyUniqueName/stock/:stockUniqueName/details?userType=:userType', // GET call
     INVENTORY_ITEM_WISE_EXPORT: 'company/:companyUniqueName/inventory/export-balance-report/item-wise?from=:from&to=:to',
+    INVENTORY_VARIANT_WISE_EXPORT: 'company/:companyUniqueName/inventory/export-balance-report/variant-wise?from=:from&to=:to',
+    INVENTORY_GROUP_WISE_EXPORT: 'company/:companyUniqueName/inventory/export-balance-report/group-wise?from=:from&to=:to',
+    INVENTORY_TRANSACTION_EXPORT: 'company/:companyUniqueName/inventory/export-balance-report/transaction-wise?from=:from&to=:to',
     INVENTORY_ADJUST: {
-        REPORT: 'company/:companyUniqueName/inventory-adjustment/adjustment-report?page=:page&from=:from&to=:to&count=:count&sortBy=:sortBy&sort=:sort&q=:q&searchBy=:searchBy&inventoryType=:inventoryType', // POST call
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/apiurls/inventory.api.ts:109]: ye q remove kiya?

File: apps/web-giddh/src/app/services/apiurls/inventory.api.ts [hunk 38]
```diff
@@ -105,16 +105,16 @@ export const INVENTORY_API = {
...
+        REPORT: 'company/:companyUniqueName/inventory-adjustment/adjustment-report?page=:page&from=:from&to=:to&count=:count&sortBy=:sortBy&sort=:sort&q=:q&searchBy=:searchBy', // POST call
         DELETE: 'company/:companyUniqueName/inventory-adjustment/:referenceNo?voucherVersion=2', // DELETE call
-        GET_REASON: 'company/:companyUniqueName/inventory-adjustment/reason/get-all?count=2000',// GET call
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/apiurls/inventory.api.ts]: ye q change kiya?

File: apps/web-giddh/src/app/services/apiurls/logs.api.ts [hunk 39]
```diff
@@ -1,5 +1,5 @@
 export const LOGS_API = {
     AUDIT_LOGS: 'company/:companyUniqueName/logs?page=:page', // post call
     GET_AUDIT_LOG_FORM_FILTERS: 'ui/auditlogs-form',
-    GET_AUDIT_LOGS_V2: 'v2/company/:companyUniqueName/logs?page=:page', // post call
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/apiurls/logs.api.ts]: ye q remove kiya?

File: apps/web-giddh/src/app/services/general.service.ts [hunk 40]
```diff
@@ -1069,13 +1069,14 @@ export class GeneralService {
                 balanceDueAmountForCompany = Number(item.totalBalance.amountForCompany) || 0;
                 balanceDueAmountForAccount = Number(item.totalBalance.amountForAccount) || 0;
             }
-            if ([VoucherTypeEnum.sales, VoucherTypeEnum.creditNote, VoucherTypeEnum.debitNote, VoucherTypeEnum.purchase, VoucherTypeEnum.receipt, VoucherTypeEnum.payment]?.indexOf(selectedVoucher) > -1 && item.grandTotal) {
+            if ([VoucherTypeEnum.sales, VoucherTypeEnum.creditNote, VoucherTypeEnum.debitNote, VoucherTypeEnum.purchase, VoucherTypeEnum.receipt, VoucherTypeEnum.payment, 'purchase-order']?.indexOf(selectedVoucher) > -1 && item.grandTotal) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/general.service.ts]: use enum for purchase order

File: apps/web-giddh/src/app/shared/bulk-export-voucher/bulk-export-voucher.component.ts [hunk 41]
```diff
@@ -8,13 +8,15 @@ import { ReplaySubject } from 'rxjs';
...
+import { VoucherComponentStore } from '../../vouchers/utility/vouchers.store';
 
 type bulkExportVoucherTypes = 'sales' | 'debit note' | 'credit note' | 'purchase' | 'payment' | 'receipt';
 
 @Component({
     selector: 'bulk-export-voucher',
     templateUrl: './bulk-export-voucher.component.html',
-    styleUrls: [`./bulk-export-voucher.component.scss`]
+    styleUrls: [`./bulk-export-voucher.component.scss`],
+    providers:[VoucherComponentStore]
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/bulk-export-voucher/bulk-export-voucher.component.ts]: remove provider

File: apps/web-giddh/src/app/shared/header/header.component.html [hunk 42]
```diff
@@ -753,13 +753,7 @@ <h3 class="modal-title pull-left">{{ localeData?.ready_to_upgrade }}</h3>
             (closeAsideEvent)="toggleSidebarPane(false, false)"
         ></aside-setting>
     </div>
-    <div class="help-support-aside setting-sidebar-wrapper" *ngIf="asideInventorySidebarMenuState === 'in'">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/header.component.html:756]: check this

File: apps/web-giddh/src/app/shared/header/header.component.scss [hunk 43]
```diff
@@ -353,153 +353,7 @@ ul.sub-list {
 .setting-sidebar {
     box-shadow: -2px -2px 15px --color-primary-common-box-shadow;
 }
-.desktop-header .hamburger-menu {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/header.component.scss:356]: check this

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 44]
```diff
@@ -1487,17 +1484,11 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
             } else if (
                 document.getElementsByTagName("tabset") &&
                 document.getElementsByTagName("tabset").length > 0 &&
-                !this.router.url.includes("/vendor") && (!document.getElementsByClassName("static-tabs-on-page") || (document.getElementsByClassName("static-tabs-on-page") && document.getElementsByClassName("static-tabs-on-page").length === 0))) {
+                !this.router.url.includes("/vendor")) {
                 document.querySelector('body').classList.add('page-has-tabs');
                 document.querySelector('body').classList.remove('on-setting-page');
                 document.querySelector('body').classList.remove('on-user-page');
                 document.querySelector('body').classList.remove('mobile-setting-sidebar');
-            }
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/header.component.ts:1495]: check this file changes

File: apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts [hunk 45]
```diff
@@ -496,7 +496,6 @@ export class PrimarySidebarComponent implements OnInit, OnChanges, OnDestroy {
             item.time = +new Date();
             let entity = (item.type) === 'MENU' ? 'menus' : 'accounts';
             this.doEntryInDb(entity, item, fromInvalidState);
-            this.closeAccountModal(true);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts:499]: check this

File: apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.html [hunk 46]
```diff
@@ -1,83 +1,214 @@
...
+                ></textarea>
+                <ng-container *ngIf="bulkUpdateForm.get('message2')?.value && !showNotesAtLastPage">
+                    <span class="max-character-label">{{ commonLocaleData?.app_characters_left }}:&nbsp;</span>
+                    <span class="max-character-value">{{ 200 - bulkUpdateForm.get("message2")?.value?.length }}</span>
+                </ng-container>
+            </mat-form-field>
+            <!-- due-date-click -->
+            <giddh-datepicker
+                *ngIf="bulkUpdateForm.get('selectedField')?.value === 'dueDate'"
+                type="text"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.html]: remove type

File: apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.ts [hunk 47]
```diff
@@ -1,15 +1,491 @@
...
+                        this.bulkUpdateForm.get("imageSignatureUniqueName")?.patchValue(response.uniqueName);
+                    }
+                } else {
+                    this.bulkUpdateForm.get("imageSignatureUniqueName")?.patchValue("");
+                }
+            });
+        }
+
+
+
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.ts]: remove extra space

File: apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts [hunk 48]
```diff
@@ -1,15 +1,77 @@
...
 
+  /**
+   * Set Purchase Order Value
+   *
+   * @private
+   * @param {*} voucher
+   * @return {*}  {FormGroup}
+   * @memberof ConvertBillDialogComponent
+   */
+  private getPurchaseOrderValue(voucher?: any): FormGroup {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts]:
```suggestion
  private getPurchaseOrderFormGroup(voucher?: any): FormGroup {
```

File: apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts [hunk 49]
```diff
@@ -1,15 +1,77 @@
...
+  /** Hold Form group */
+  public form: FormGroup;
 
-  constructor() { }
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public vouchers,
+    public dialogRef: MatDialogRef<any>,
+    private formBuilder: FormBuilder
+  ) {
+    this.initForm();
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts]: format code with 4 spaces

File: apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts [hunk 50]
```diff
@@ -90,7 +122,9 @@ export class VoucherComponentStore extends ComponentStore<VoucherState> {
         private ledgerService: LedgerService,
         private commonService: CommonService,
         private accountService: AccountService,
-        private searchService: SearchService
+        private searchService: SearchService,
+        private bulkUpdateInvoiceService: InvoiceBulkUpdateService,
+        private bulkVoucherExportService: BulkVoucherExportService
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts]: move methods to voucher service

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/services/apiurls/inventory.api.ts (hunk 37 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/services/apiurls/inventory.api.ts:109]: revert

File: apps/web-giddh/src/app/services/apiurls/logs.api.ts (hunk 39 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/services/apiurls/logs.api.ts]: revert

File: apps/web-giddh/src/app/shared/header/header.component.html (hunk 42 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/shared/header/header.component.html:756]: remove from test

File: apps/web-giddh/src/app/shared/header/header.component.scss (hunk 43 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/shared/header/header.component.scss:356]: remove this on test

File: apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts (hunk 45 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts:499]: remove this on test

File: apps/web-giddh/src/app/shared/header/header.component.ts (hunk 44 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/shared/header/header.component.ts:1495]: Inform me when there are changes to the test.

**Comment by dvCodeWorld** [comment]
Infected Branch hence closed this pr

**Pull request closed without merging**